make/protogen.mk

```diff
@@ -37,15 +37,16 @@ endif
 PROTOC_VERSION := 21.4
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
-PROTOC_ARCH = linux
+PROTOC_OS = linux
 endif
 ifeq ($(UNAME_S),Darwin)
-PROTOC_ARCH = osx
+PROTOC_OS = osx
 endif
+PROTOC_ARCH=$(shell case $$(uname -m) in (aarch64) echo armv6 ;; (*) uname -m ;; esac)
```

Comment:
Not sure I'm following this logic? This should move arm64 to aarch64 right?

Reply:
This broke building on m1 macs

Reply:
at least for downloading protoc

Reply:
Sorry for the inconvenience @connorgorman .

Interesting, because previously line 64 was hardcoded to `x86_64` regardless of runtime CPU architecture.
Are you saying that `x86_64` tarball actually works on M1?? Or did you just happen to have an M1-compatible binary stashed away manually at `$(PROTO_PRIVATE_DIR)/protoc-$(PROTOC_ARCH)-$(PROTOC_VERSION)` and now this makefile changed to `$(PROTO_PRIVATE_DIR)/protoc-$(PROTOC_OS)-$(PROTOC_ARCH)-$(PROTOC_VERSION)` and your local copy is no longer used?

Can you please provide more information about how it's actually broken now?

The reason I chose this specific recipe for converting `uname -m` to download file infix is by comparing the download directories for `protoc` to the one for `kuttl`, which I assume works because nobody ever complained :wink:

Reply:
Anyway, this should be solved by #4370

Reply:
Ah! it probably did work on m1 with amd64 because I would assume that rosetta did its magic :)